src/app/clusters/closure-control-server/closure-control-cluster-logic.h

```diff
@@ -183,30 +172,215 @@ class ClusterLogic
     CHIP_ERROR GetMainState(MainStateEnum & mainState);
     CHIP_ERROR GetOverallState(DataModel::Nullable<GenericOverallState> & overallState);
     CHIP_ERROR GetOverallTarget(DataModel::Nullable<GenericOverallTarget> & overallTarget);
+    // The delegate is expected to return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED to indicate end of list
+    CHIP_ERROR GetCurrentErrorList(const AttributeValueEncoder::ListEncodeHelper & aEncoder);
 
-    // TODO: Add ErrorList guetter with its implementation
+    /**
+     * @brief Set OverallTarget.
+     *
+     * @param[in] overallTarget OverallTarget Position, Latch and Speed.
+     *
+     * @return CHIP_NO_ERROR if set was successful.
+     *         CHIP_ERROR_INCORRECT_STATE if the cluster has not been initialized.
+     *         CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE if feature is not supported.
+     *         CHIP_ERROR_INVALID_ARGUMENT if argument are not valid
+     */
+    CHIP_ERROR SetOverallState(const DataModel::Nullable<GenericOverallState> & overallState);
 
-    // All Set functions
-    // Return CHIP_ERROR_INCORRECT_STATE if the class has not been initialized.
-    // Return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE if the attribute is not supported by the conformance.
-    // Return CHIP_ERROR_INVALID_ARGUMENT if the input value is out of range.
-    // Returns CHIP_ERROR_PERSISTED_STORAGE_FAILED if the value could not not be stored in persistent storage.
-    // Otherwise return CHIP_NO_ERROR and set the parameter value in the cluster state
-    // Set functions are supplied for any values that can be set either internally by the device or externally
-    // through a direct attribute write. Changes to attributes that happen as a side effect of cluster commands
-    // are handled by the cluster command handlers.
-
-    CHIP_ERROR SetCountdownTime(const DataModel::Nullable<ElapsedS> & countdownTime);
+    /**
+     * @brief Function to change the MainState.
+     *
+     *        See SetCountdownTime function comment below
+     */
     CHIP_ERROR SetMainState(MainStateEnum mainState);
-    CHIP_ERROR SetOverallState(const DataModel::Nullable<GenericOverallState> & overallState);
+
+    /**
+     * @brief Public API to trigger countdown time update from the delegate (application layer).
+     *        Function calls the SetCountdownTime function with the fromDelegate parameter set to true.
```

Comment:
This is an implementation detail, not API.

API documentation would describe why/when the caller might want to call this.